src/storage/src/source/generator.rs

```diff
@@ -40,6 +40,14 @@ pub struct LoadGeneratorSourceReader {
     tick: Duration,
     offset: MzOffset,
     pending: Vec<Row>,
+    // Load-generator sources support single-threaded ingestion only, so only
+    // one of the `LoadGeneratorSourceReader`s will actually produce data.
+    active_read_worker: bool,
+    // The non-active reader (see above `active_read_worker`) has to report back
+    // that is is not consuming from the one [`PartitionId:None`] partition.
+    // Before it can return a [`NextMessage::Finished`]. This is keeping track
+    // of that.
+    reported_unconsumed_partitions: bool,
```

Comment:
this never seems to be set to `true`?

Reply:
Oh boy, good catch! This wasn't caught because we never run sources with more than one timely worker. Good thing that this removes barriers to doing that. I think @philip-stoev is eager to enable multi-worker sources in tests. :sweat_smile: